### ISSUE TITLE
ListView reversed `Return to top` header tap

### ISSUE THREAD

**Issue by vasilich6107** [flutter/flutter#83397]
## Use case
I'm building a chat like interface and I'm using `ListView` with `reversed` setted to `true` to make UI look "chat like"
The issue I'm experiencing is that `return to top` behaviour is also reversed. 
In this case is scrolls the chat down to index `0` item. As expected by design. 
But I would like to have an ability to hook into this behaviour or any way to modify it's behaviour.

What I need is to modify `return to top` behaviour to make it scroll to the last item in `reversed` `ListView`.

I inspected several public packages which implement "chat like" UI and they have the same issues as described here.

I googled for several hours and didn't find any easy or suitable solution for my issue.

## Proposal
Some way to hook into `return to top` or any way to modify it's behaviour.

**Comment by AbhishekDoshi26** [comment]
@vasilich6107 You can provide `_scrollController` to the ListView and then use something like this:
```
 _scrollController.animateTo(
            0.0,
            curve: Curves.easeOut,
            duration: const Duration(milliseconds: 300),
          );
```
Change `0.0` to any value that you wish to reach.

**Comment by vasilich6107** [comment]
@AbhishekDoshi26 I'm not sure that your solution fits my needs
I need to alter scrollcontroller behavior on "header tap" in iOs

**Comment by AbhishekDoshi26** [comment]
Yes, so you can setState the above solution onTap of your header
@vasilich6107

**Comment by vasilich6107** [comment]
But I'm requesting an easier way to hook into Scaffold header tap)

**Comment by vasilich6107** [comment]
The possible workaround is:
- wrap your `Scaffold` in `Stack` with 
```dart
GestureDetector(
          behavior: HitTestBehavior.opaque,
          child: SizedBox(
            width: double.infinity,
            height: MediaQuery.of(context).padding.top,
          ),
          onTap: () {
            print('to home');
          },
        )
```
- pass into context your custom `ChatScrollController`(the same way as `PrimaryScrollController` does)
- get your scroll controller via `ChatScrollController.of(context)` and this is it
